onboarding_project/squad_pantry_app/admin.py

```diff
@@ -22,11 +31,150 @@ class OrderDishInline(admin.StackedInline):
+    exclude = ('placed_by', )
+    readonly_fields = ('closed_at', )
+    list_display = ('placed_by', 'status', 'created_at', 'scheduled_time')
+
+    def has_add_permission(self, request):
+        return not request.user.is_kitchen_staff
+
+    def get_urls(self):
+        return [
+            url(
+                r'^(.+)/change/cancel-order/$',
+                self.admin_site.admin_view(self.cancel_order_view),
+                name='cancel_order_view',
+            ),
+        ] + super(OrderAdmin, self).get_urls()
+
+    def cancel_order_view(self, request, object_id, extra_context=None):
+        try:
+            order = Order.objects.get(pk=object_id)
+        except Order.DoesNotExist:
+            self.message_user(request, 'Mentioned Order ID does not exists', messages.ERROR)
+            return HttpResponseRedirect(request.META.get('HTTP_REFERER', '/admin/squad_pantry_app/order'))
+        else:
+            cancelled = order.cancel_order(request.user.id)
+            if cancelled == order.CANCEL_SUCCESS:
+                self.message_user(request, 'Order cancelled', messages.SUCCESS)
+            elif cancelled == order.WRONG_USER:
+                self.message_user(request, 'You do not have permission to cancel this order', messages.ERROR)
+            elif cancelled == order.PROCESSING_ORDER:
+                self.message_user(request, 'Could not cancel, The order is already being processed', messages.ERROR)
+            elif cancelled == order.ORDER_CLOSED_ERROR:
+                self.message_user(request, 'The order is already closed', messages.ERROR)
+
+            return HttpResponseRedirect(request.META.get('HTTP_REFERER', '/'))
+
+    def save_model(self, request, obj, form, change):
+        if not change:
+            obj.placed_by = request.user
+        return super(OrderAdmin, self).save_model(request, obj, form, change)
+
+    def get_readonly_fields(self, request, obj=None):
+        user = request.user
+        if obj is None:
+            if not user.is_kitchen_staff or (user.is_kitchen_staff and obj.status in obj.CLOSED_ORDERS
+                                             and obj.closed_at is not None):
+                return self.readonly_fields + ('status', )
+            return self.readonly_fields
+        else:
+            if not user.is_kitchen_staff or (user.is_kitchen_staff and obj.status in obj.CLOSED_ORDERS
+                                             and obj.closed_at is not None):
+                return self.readonly_fields + ('status', 'scheduled_time')
+            return self.readonly_fields + ('scheduled_time', )
+
+    def get_queryset(self, request):
+        qs = super(OrderAdmin, self).get_queryset(request)
+        if not request.user.is_kitchen_staff:
+            return qs.filter(placed_by=request.user)
+        return qs.filter()
+
+
+class UserCreationForm(forms.ModelForm):
+    password1 = forms.CharField(label='Password', widget=forms.PasswordInput())
```

Comment:
Rename to `password` and `password_confirmation`. Let's use easy-to-read variable names

Reply:
Done!